security/s2a/internal/crypter/counter.go

```diff
@@ -0,0 +1,42 @@
+package crypter
+
+import "errors"
+
+const counterLen = 12
+
+// counter is a 96-bit, little-endian counter.
+type counter struct {
+	value   [counterLen]byte
+	invalid bool
+}
+
+func newCounter() counter {
```

Comment:
Please allow the caller to pass in the sequence number as an argument. (In our use cases, the sequence number will always start at zero, but this may change in the future.)

Reply:
Done